### PR TITLE
Replace NV shader stage enums with KHR version in the glsl compiler, add missing ray tracing shader stages

### PR DESCRIPTION
**Pull request by SaschaWillems**
## Description

This PR replaces the the NV enums for shader stages used in the framework's glsl compiler. As the NV and KHR enums have the same values, this does not affect functionality and is more of a cosmetic clean-up.

It also adds the the missing shader stages for callable, intersection and anyhit ray tracing shaders.

## General Checklist:

Please ensure the following points are checked:

- [x] My code follows the [coding style](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#Code-Style)
- [x] I have reviewed file [licenses](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#Copyright-Notice-and-License-Template)
- [x] I have updated the header Copyright to reflect the current year (CI build will fail if Copyright is out of date)
- [x] I have commented any added functions (in line with Doxygen)
- [x] I have commented any code that could be hard to understand
- [x] My changes do not add any new compiler warnings
- [x] My changes do not add any new validation layer errors or warnings
- [x] I have used existing framework/helper functions where possible
- [x] My changes build on Windows, Linux, macOS and Android. Otherwise I have [documented any exceptions](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#General-Requirements)
- [x] My changes do not add any regressions
- [x] I have tested every sample to ensure everything runs correctly
- [x] This PR describes the scope and expected impact of the changes I am making

### CODE REVIEW

**Review by gpx1000 — DISMISSED** [review]
Yep, exactly what should be changed.

File: framework/common/vk_common.cpp [hunk 1]
```diff
@@ -96,15 +96,19 @@ VkShaderStageFlagBits find_shader_stage(const std::string &ext)
...
 	}
 	else if (ext == "rchit")
 	{
-		return VK_SHADER_STAGE_CLOSEST_HIT_BIT_NV;
+		return VK_SHADER_STAGE_CLOSEST_HIT_BIT_KHR;
+	}
+	else if (ext == "rcall")
+	{
+		return VK_SHADER_STAGE_CALLABLE_BIT_KHR;
 	}
```
Comment (mklefrancois) [inline framework/common/vk_common.cpp:121]: For completion, there should be the support for "rahit"  -> VK_SHADER_STAGE_INTERSECTION_BIT_KHR

File: framework/glsl_compiler.cpp [hunk 2]
```diff
@@ -51,14 +51,17 @@ inline EShLanguage FindShaderLanguage(VkShaderStageFlagBits stage)
...
+		case VK_SHADER_STAGE_MISS_BIT_KHR:
+			return EShLangMiss;
 
-		case VK_SHADER_STAGE_CLOSEST_HIT_BIT_NV:
-			return EShLangClosestHitNV;
+		case VK_SHADER_STAGE_CLOSEST_HIT_BIT_KHR:
+			return EShLangClosestHit;
+
+		case VK_SHADER_STAGE_CALLABLE_BIT_KHR:
+			return EShLangCallable;
```
Comment (mklefrancois) [inline framework/glsl_compiler.cpp:71]: For completion, there should be also the case for VK_SHADER_STAGE_INTERSECTION_BIT_KHR

**Review by mklefrancois — DISMISSED** [review]
All good, as long as all samples are also using the KHR for ray tracing.

File: framework/common/vk_common.cpp (hunk 1 above)
Comment (SaschaWillems) [reply framework/common/vk_common.cpp:121]: Good point. I have added all missing ray tracing shader stages to the PR, including any hit and intersection stages.

File: framework/glsl_compiler.cpp (hunk 2 above)
Comment (SaschaWillems) [reply framework/glsl_compiler.cpp:71]: Good point. I have added all missing ray tracing shader stages to the PR, including any hit and intersection stages.

**Review by gpx1000 — APPROVED** [review]

**Review by r-potter — APPROVED** [review]

**Review by mklefrancois — APPROVED** [review]

**Pull request merged**